src/vmspawn/vmspawn.c

```diff
@@ -1173,6 +1198,74 @@ static int generate_ssh_keypair(const char *key_path, const char *key_type) {
         return 0;
 }
 
+static int generate_mac(const char *machine_name, struct ether_addr *mac, sd_id128_t hash_key, uint64_t idx) {
```

Comment:
I guess this is basically copied from nspawn-network.c. Maybe, better to expose it to e.g. netif-util.[ch] and reuse it.